var/spack/repos/builtin/packages/amrex/package.py

```diff
@@ -341,8 +341,13 @@ def test(self):
         args = []
         args.append("-S./cache/amrex/Tests/SpackSmokeTest")
         args.append("-DAMReX_ROOT=" + self.prefix)
-        args.append("-DMPI_C_COMPILER=" + self.spec["mpi"].mpicc)
-        args.append("-DMPI_CXX_COMPILER=" + self.spec["mpi"].mpicxx)
+        if "+mpi" in self.spec:
+            args.append("-DMPI_C_COMPILER=" + self.spec["mpi"].mpicc)
+            args.append("-DMPI_CXX_COMPILER=" + self.spec["mpi"].mpicxx)
+
+        if "+cuda" in self.spec:
+            args.append("-DCMAKE_CUDA_COMPILER=" + self.spec["cuda"].prefix.bin + "/nvcc")
```

Comment:
I think `join_path` would be better than `+ "/..."`

Reply:
Made the change, thanks for the feedback! :smile: